app/scripts/metamask-controller.js

```diff
@@ -322,6 +323,13 @@ export default class MetamaskController extends EventEmitter {
     if (chrome.braveWallet.hasOwnProperty('ready')) { // eslint-disable-line no-undef
       chrome.braveWallet.ready() // eslint-disable-line no-undef
     }
+
+    // if it is Brave Browser and chrome.ipfs flag is available then use native Brave ipfs
+    getIPFSEnabledFlag().then((isBraveIpfs) => {
+      if (isBraveIpfs) {
+        this.setIpfsGateway('ipfs://<cid>')
```

Comment:
I don't think this will be handled properly when resolving ENS because you aren't replacing <cid> with the cid that's retrieved.

Reply:
I think this also doesn't use setIpfsGateway correctly because that takes 2 params and the second param is a callback.  So if this gets hit it will error out.